src/Objects/DeviceFlag.vala

```diff
@@ -0,0 +1,363 @@
+        }
+        if ((flags & (1u << 30)) > 0) {
+            list.append (DeviceFlag.VERSION_CHECK_REQUIRED);
+        }
+        if ((flags & (1u << 31)) > 0) {
+            list.append (DeviceFlag.INSTALL_ALL_RELEASES);
+        }
+        if ((flags & (1u << 32)) > 0) {
+            list.append (DeviceFlag.MD_SET_NAME);
+        }
+        if ((flags & (1u << 33)) > 0) {
+            list.append (DeviceFlag.MD_SET_NAME_CATEGORY);
+        }
+        if ((flags & (1u << 34)) > 0) {
+            list.append (DeviceFlag.MD_SET_VERFMT);
+        }
+        if ((flags & (1u << 35)) > 0) {
+            list.append (DeviceFlag.ADD_COUNTERPART_GUIDS);
+        }
+        if ((flags & (1u << 36)) > 0) {
+            list.append (DeviceFlag.NO_GUID_MATCHING);
+        }
+        if ((flags & (1u << 37)) > 0) {
+            list.append (DeviceFlag.UPDATABLE_HIDDEN);
+        }
+        if ((flags & (1u << 38)) > 0) {
+            list.append (DeviceFlag.SKIPS_RESTART);
+        }
+        if ((flags & (1u << 39)) > 0) {
+            list.append (DeviceFlag.HAS_MULTIPLE_BRANCHES);
+        }
+        if ((flags & (1u << 40)) > 0) {
+            list.append (DeviceFlag.BACKUP_BEFORE_INSTALL);
+        }
+        if ((flags & (1u << 41)) > 0) {
+            list.append (DeviceFlag.MD_SET_ICON);
+        }
+        if ((flags & (uint64.MAX)) > 0) {
+            list.append (DeviceFlag.UNKNOWN);
+        }
+
+        return list;
+    }
+
+    // https://gitlab.gnome.org/hughsie/gnome-firmware-updater/-/blob/f5281078e3cfade7ff919c812ba63de22431aaf2/src/gfu-common.c#L249
+    public string? to_string () {
+        switch (this) {
+            case NONE:
+                return null;
+            case INTERNAL:
+                /* TRANSLATORS: Device cannot be removed easily*/
+                return _("Internal device");
+            case UPDATABLE:
+                /* TRANSLATORS: Device is updatable in this or any other mode */
+                return _("Updatable");
+            case ONLY_OFFLINE:
+                /* TRANSLATORS: Update can only be done from offline mode */
+                return _("Update requires a reboot");
+            case REQUIRE_AC:
+                /* TRANSLATORS: Must be plugged in to an outlet */
+                return _("System requires external power source");
+            case LOCKED:
+                /* TRANSLATORS: Is locked and can be unlocked */
+                return _("Device is locked");
+            case SUPPORTED:
+                /* TRANSLATORS: Is found in current metadata */
+                return _("Supported on LVFS");
+            case NEEDS_BOOTLOADER:
+                /* TRANSLATORS: Requires a bootloader mode to be manually enabled by the user */
+                return _("Requires a bootloader");
+            case NEEDS_REBOOT:
+                /* TRANSLATORS: Requires a reboot to apply firmware or to reload hardware */
+                return _("Needs a reboot after installation");
```

Comment:
Same as above; is there any reason this should be translated differently? The end result is that it will require a restart.

```suggestion
                return _("Update requires restarting the system");
```